### PR TITLE
[WIP] sets up initial sunspot stuff

### PR DESCRIPTION
**Pull request by straleyb**
This is a PR which sets up sunspot for searching in CVM. 

Sunspot is a solr powered lightweight search mechanism for Rails. https://github.com/sunspot/sunspot

Sunspot uses solr, which is a technology we already use in the stack. You can configure models to be able to be searchable based on attributes on that model. 

One initial concern I have and something Im currently working through is, since CVM doesnt use attributes on models, but properties, it doesnt make it exactly one to one when mapping fields. This could be an initial hurdle that is seeming to take a bit more engineering so that we can pull text from the properties instead of attributes. 

Sunspot is very powerful and a lot of discussion could be had about how we want to weight searches and full text results based on important attributes. Overall, this may or may not be a viable options, I still have to see depending on what needs to be done with the properties.

### CODE REVIEW

File: app/controllers/search_controller.rb [hunk 1]
```diff
@@ -3,6 +3,8 @@ def index
   end
 
   def search_results
-    Vocabulary.search{ fulltext params[:search_value]}
+    search = Vocabulary.search{ fulltext params[:search_value]}
+    puts search.results
```
Comment (revgum) [inline app/controllers/search_controller.rb]: Replace with Rails.logger or eliminate puts

File: app/injectors/vocabulary_injector.rb [hunk 2]
```diff
@@ -12,6 +12,7 @@ def all_vocabs_query
     #causes some issues with the objects and predicates being fetched properly
     #By inputing the sparql_client directly into the all vocabs query correctly
     #assembles the query.
+    #
```
Comment (revgum) [inline app/injectors/vocabulary_injector.rb]: unnecessary?

**Comment by straleyb** [comment]
So with this branch. IDs are searchable. Im having trouble getting all the other fields to index properly. I had to make some changes to fit the way Sunspot is expecting as an interface to a model, but it should be working now. 

In order to make this work, ingest a vocabulary with a title, then in the console run `rake cvm:reindex[0,Vocabulary]` Then you can go to the search page and search for the ID of the item you ingested. This will display the vocabularies title. This is a quick hack and we need to discuss whether or not this is worth it. Like I said, I havent been able to get the other fields to work, but ids are indexing properly.

**Comment by revgum** [comment]
invalid PR, replaced with work by Ryan using Sunspot

**Pull request closed without merging**